src/main/java/org/jenkinsci/plugins/docker/commons/tools/DockerToolInstaller.java

```diff
@@ -0,0 +1,118 @@
+
+    @DataBoundConstructor
+    public DockerToolInstaller(String label, String version) {
+        super(label);
+        this.version = version;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    @Override
+    public FilePath performInstallation(ToolInstallation toolInstallation, Node node, TaskListener listener) throws IOException, InterruptedException {
+
+        String os = node.getChannel().call(new Callable<String, IOException>() {
+            public String call() throws IOException {
+                String os = System.getProperty("os.name").toLowerCase();
+                String arch = System.getProperty("os.arch").contains("64") ? "x86_64" : "i386";
+                if(os.contains("linux"))  return "Linux/"+arch;
+                if(os.contains("windows"))   return "Windows/"+arch;
+                if(os.contains("mac"))   return "Darwin/"+arch;
+                throw new IOException("Failed to determine OS architecture "+os+":"+arch);
+            }
+        });
+
+        URL url = new URL("https://get.docker.com/builds/"+os+"/docker-"+version);
+        FilePath install = preferredLocation(tool, node);
+
+        // (simplified) copy/paste from FilePath as hudson.FilePath.installIfNecessaryFrom do assume the URL points to be a zip/tar archive
+
+        FilePath timestamp = install.child(".timestamp");
+        URLConnection con;
+        try {
+            con = ProxyConfiguration.open(url);
+            if (timestamp.exists()) {
+                con.setIfModifiedSince(timestamp.lastModified());
+            }
+            con.connect();
+
+        } catch (IOException x) {
+            if (install.exists()) {
+                // Cannot connect now, so assume whatever was last unpacked is still OK.
+                if (listener != null) {
+                    listener.getLogger().println("Skipping installation: " + x);
+                }
+                return install;
+            } else {
+                throw x;
+            }
+        }
+
+        if (con instanceof HttpURLConnection
+                && ((HttpURLConnection)con).getResponseCode() == HttpURLConnection.HTTP_NOT_MODIFIED) {
+            return install;
+        }
+
+        listener.getLogger().println("Downloading Docker client " + version);
+
+        long sourceTimestamp = con.getLastModified();
+
+        if (install.exists()) {
+            if(timestamp.exists() && sourceTimestamp ==timestamp.lastModified())
+                return install;   // already up to date
+            install.deleteContents();
+        }
+
+        FilePath bin = install.child("bin");
+        bin.mkdirs();
+        FilePath docker = bin.child("docker");
+        docker.copyFrom(ProxyConfiguration.getInputStream(url));
+        docker.act(new FilePath.FileCallable<Object>() {
+            public Object invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
+                f.setExecutable(true);
```

Comment:
This is Java 6+. Are we not enforcing Animal Sniffer here? Maybe the OSS plugin parent POM omits that. Last I checked it was enforcing Java 5-. (Yes we want to assume 6+, I was just surprised that this compiles as is.)